src/animations.ts

```diff
@@ -111,105 +455,107 @@ let FAILED_KARAOKE_UPDATE_ATTEMPTS = 0
-    const animation = ANIMATIONS[HERO_VIDEO_ANIMATION]
-
-    setRadialProgressBar(animation, HERO_VIDEO.currentTime * 1000)
+function sampleQuestionAnimation(animationName: AnimationName, karaokeState: KaraokeState = null) {
+    const animation = ANIMATIONS[animationName]
+    const isSameAudio = (PLAYER.querySelector('source').src === animation.expectedAudioSrc)
+    const isSameAudioPlaying = isSameAudio && !PLAYER.paused
 
-    window.requestAnimationFrame(heroAnimation)
+    if (isSameAudioPlaying) {
+        const animationTime = PLAYER.currentTime * 1000
+        setRadialProgressBar(animation, animationTime)
+        karaokeState = attemptUpdateKaraoke(animation.karaoke, animationTime, karaokeState)
+        window.requestAnimationFrame(() => sampleQuestionAnimation(animationName, karaokeState))
+    } else if (!isSameAudio) {
+        // Another audio file has started playing, so this animation can be reset.
+        animation.cleanupAnimation()
+    }
 }
 
 
-function yourLittleOneAnimation() {
-    if (CURRENT_ANIMATION_INFO.name === YOUR_LITTLE_ONE_ANIMATION) {
-        const animation = ANIMATIONS[YOUR_LITTLE_ONE_ANIMATION]
-        const isAudioPlaying = !PLAYER.paused && (PLAYER.querySelector('source').src === animation.expectedAudioSrc)
-
-        // animationTime represents the current progress into the animation.
-        // If the appropriate audio is playing, animationTime is the number of milliseconds from the player's
-        // currentTime.
-        // Otherwise, animationTime is the number of milliseconds since the animation was scrolled into view (and using
-        // the modulo operator to account for looping animations).
-        const animationTime = (
-            isAudioPlaying ?
-                PLAYER.currentTime * 1000 :
-                ((new Date()).valueOf() - CURRENT_ANIMATION_INFO.timeScrolledIntoView) % animation.duration
-        )
-
-        for (const step of animation.steps) {
-            const element = document.getElementById(step.id)
-            if (animationTime >= step.start && animationTime < step.end) {
-                element.style.opacity = '1'
-            } else {
-                element.style.opacity = '0'
-            }
-        }
+// Set the animation back to the initial state.
+function sampleQuestionAnimationCleanup(animationName: AnimationName) {
+    const animation = ANIMATIONS[animationName]
+    setRadialProgressBar(animation, 0)
+    attemptUpdateKaraoke(animation.karaoke, 0, null)
+}
 
-        setRadialProgressBar(animation, animationTime)
-        attemptUpdateKaraoke(animation.karaoke, animationTime)
 
-        window.requestAnimationFrame(yourLittleOneAnimation)
-    }
-}
+let FAILED_GET_SAMPLE_QUESTION_COMPONENTS_ATTEMPTS = 0
+
+// Since the sample questions are created programmatically from the CMS, they may not be in the DOM by the time this
+// script runs (even with `defer`). Attempt to fetch the components for 10 seconds before sending a Sentry alert.
```

Comment:
It is a reasonable approach 👍 
Does webflow expose their JS they use to load the data from the CMS? If so we could put the code there and use a `.then()`.
I am wondering because you can export all the code in Webflow and host it yourself, thus this JS code must be accessible, but maybe not with the hosted version :-)

Reply:
I'm going to update this comment, I was mistaken on what's causing this delay. The CMS content is in the HTML already, however there's a library we use to moves the CMS elements into the slider, and that's why there was a delay.

Instead of searching from `sampleQuestionsSlider` to find these components, I think we could search from a common parent to find these elements. If that works, it won't matter if the elements have been moved yet and this shouldn't need to run multiple times 😄  Will do some testing to find out more!

If that doesn't work, do have the minified script that's used to move the contents. Their repo isn't public, so not sure if we can get the source code.

Reply:
I've revised my selector and I still sometimes see a failed attempt before it can find the elements. I think for v1 what we have now (with my latest commits) is probably okay. Let me know if you think otherwise!